src/main/java/org/spoken_tutorial/health/elasticsearch/contentfile/ContentsfromFile.java

```diff
@@ -19,21 +26,119 @@
 @Component
 public class ContentsfromFile {
 
+    private static final Logger logger = LoggerFactory.getLogger(ContentsfromFile.class);
+
     @Autowired
     private Config config;
 
+    private String handleZipFile(final Parser parser, InputStream inputStream) {
+
+        ZipEntry zipEntry;
+        String content = "";
+        int count = 0;
+        BodyContentHandler handler = new BodyContentHandler(10000000);
+        Metadata metadata = new Metadata();
+        List<String> allowedExtensions = Arrays.asList(config.ALLOWED_EXTENSIONS.split(","));
+
+        try (ZipInputStream zis = new ZipInputStream(inputStream)) {
+            while ((zipEntry = zis.getNextEntry()) != null) {
+
+                if (zipEntry.isDirectory()) {
+                    continue;
+                }
+
+                String name = zipEntry.getName().toLowerCase();
+
+                if (name.endsWith(".zip")) {
+                    continue;
+                }
+                int index = name.lastIndexOf(".");
+                if (index == -1) {
+                    continue;
+                }
+                if (!allowedExtensions.contains(name.substring(index + 1))) {
+                    continue;
+                }
+                if (name.endsWith(".txt")) {
+
+                    content += new String(readFromStream(zis));
+                    count++;
+
+                } else {
+                    parser.parse(zis, handler, metadata, new ParseContext());
```

Comment:
you have to open the zipEntry and parse that and not zis. again, use try-with-resources to open.